samples/rocdecDecode/rocdecdecode.cpp

```diff
@@ -0,0 +1,751 @@
+    p_dec_info->coded_width = format->coded_width;
+    p_dec_info->coded_height = format->coded_height;
+    p_dec_info->bytes_per_pixel = bitdepth_minus_8 > 0 ? 2 : 1;
+    std::ostringstream input_video_info_str;
+    input_video_info_str.str("");
+    input_video_info_str.clear();
+    input_video_info_str << "Input Video Information" << std::endl
+        << "\tCodec        : " << format->codec << std::endl;
+        if (format->frame_rate.numerator && format->frame_rate.denominator) {
+            input_video_info_str << "\tFrame rate   : " << format->frame_rate.numerator << "/" << format->frame_rate.denominator << " = " << 1.0 * format->frame_rate.numerator / format->frame_rate.denominator << " fps" << std::endl;
+        }
+    input_video_info_str << "\tSequence     : " << (format->progressive_sequence ? "Progressive" : "Interlaced") << std::endl
+        << "\tCoded size   : [" << format->coded_width << ", " << format->coded_height << "]" << std::endl
+        << "\tDisplay area : [" << format->display_area.left << ", " << format->display_area.top << ", "
+            << format->display_area.right << ", " << format->display_area.bottom << "]" << std::endl
+        << "\tBit depth    : " << format->bit_depth_luma_minus8 + 8
+    ;
+    input_video_info_str << std::endl;
+    std::cout << input_video_info_str.str();
+    return 1;
+}
+
+int ROCDECAPI handle_picture_decode(void* user_data, RocdecPicParams* params) {
+    DecoderInfo *p_dec_info = static_cast<DecoderInfo *>(user_data);
+    CHECK(rocDecDecodeFrame(p_dec_info->decoder, params));
+    return 1;
+}
+
+int ROCDECAPI handle_picture_display(void* user_data, RocdecParserDispInfo* disp_info) {
+    DecoderInfo *p_dec_info = static_cast<DecoderInfo *>(user_data);
+    RocdecProcParams params = {};
+    params.progressive_frame = disp_info->progressive_frame;
+    params.top_field_first = disp_info->top_field_first;
+    void* dev_mem_ptr[3] = { 0 };
+    uint32_t pitch[3] = { 0 };
+    CHECK(rocDecGetVideoFrame(p_dec_info->decoder, disp_info->picture_index, dev_mem_ptr, pitch, &params));
+    p_dec_info->mem_type = OUT_SURFACE_MEM_DEV_INTERNAL;
+    // check if decoding is complete
+    RocdecDecodeStatus dec_status;
+    memset(&dec_status, 0, sizeof(dec_status));
+    CHECK(rocDecGetDecodeStatus(p_dec_info->decoder, disp_info->picture_index, &dec_status));
+    if ((dec_status.decode_status == rocDecodeStatus_Error || dec_status.decode_status == rocDecodeStatus_Error_Concealed)) {
+        std::cerr << "Decode Error occurred for picture: " << disp_info->picture_index << std::endl;
+        return 0;
+    }
+
+    if (p_dec_info->dump_decoded_frames) {
+        save_frame_to_file(p_dec_info, dev_mem_ptr, pitch);
+    }
+    return 1;
+}
+
+
+void create_parser(DecoderInfo& dec_info) {
+    RocdecParserParams params = {};
+    params.codec_type = dec_info.rocdec_codec_id;
+    params.max_num_decode_surfaces = 6;
+    params.max_display_delay = 0;
```

Comment:
Please use `params.max_display_delay = 1;` to have a display delay of 1 for better performance.